vault/resource_policy.go

```diff
@@ -77,6 +80,7 @@ func policyRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	d.Set("policy", policy)
+	d.Set("name", name)
```

Comment:
Should we set this in the `CREATE` method, `policyWrite`? Right now, `policyWrite` does not call `policyRead`, so the `name` attribute won't be set until a `refresh` operation happens. 

Normally we call `create` and it returns `read`. If that's not necessary here, I recommend we set `name` in `policyWrite` then.

Reply:
Good call, I missed that wasn't happening here. Updated to now call `policyRead` from `policyWrite`.